pyqg/parameterizations.py

```diff
@@ -0,0 +1,292 @@
+        """Whether the parameterization applies to velocity (in which case this
+        property should return :code:`"uv_parameterization"`) or potential
+        vorticity (in which case this property should return
+        :code:`"q_parameterization"`). If you inherit from
+        :code:`UVParameterization` or :code:`QParameterization`, this will be
+        defined automatically.
+
+        Returns
+        -------
+        parameterization_type : string
+            Either :code:`"uv_parameterization"` or
+            :code:`"q_parameterization"`, depending on how the output should be
+            interpreted.
+        """
+        pass
+
+    def __add__(self, other):
+        """Add two parameterizations (returning a new object).
+
+        Parameters
+        ----------
+        other : Parameterization
+            The parameterization to add to this one.
+
+        Returns
+        -------
+        sum : Parameterization
+            The sum of the two parameterizations.
+        """
+        return CompositeParameterization(self, other)
+
+    def __mul__(self, constant):
+        """Multiply a parameterization by a constant (returning a new object).
+
+        Parameters
+        ----------
+        constant : number
+            Multiplicative factor for scaling the parameterization.
+
+        Returns
+        -------
+        product : Parameterization
+            The parameterization times the constant.
+        """
+        return WeightedParameterization(self, constant)
+
+    __rmul__ = __mul__
+
+
+class CompositeParameterization(Parameterization):
+    """A sum of multiple parameterizations. Used in
+    Parameterization#__add__."""
+
+    def __init__(self, *params):
+        assert len(set(p.parameterization_type for p in params)) == 1, \
+            "all parameterizations must target the same variable (uv or q)"
+        self.params = params
+
+    @property
+    def parameterization_type(self):
+        return self.params[0].parameterization_type
+
+    def __call__(self, m):
+        return np.sum([f(m) for f in self.params], axis=0)
+
+    def __repr__(self):
+        return f"CompositeParameterization{self.params}"
+
+class WeightedParameterization(Parameterization):
+    """A weighted parameterization. Used in Parameterization#__mul__."""
+
+    def __init__(self, param, weight):
+        self.param = param
+        self.weight = weight
+
+    @property
+    def parameterization_type(self):
+        return self.param.parameterization_type
+
+    def __call__(self, m):
+        return np.array(self.param(m)) * self.weight
+
+    def __repr__(self):
+        return f"{self.weight} * {self.param}"
+
+class UVParameterization(Parameterization):
+    """A generic class representing a subgrid parameterization in terms of
+    velocity. Inherit from this to define a new velocity parameterization."""
+
+    @property
+    def parameterization_type(self):
+        ""
+        return 'uv_parameterization'
```

Comment:
```suggestion
    parameterization_type(self) = 'uv_parameterization'
```

class attribute is probably simpler here, no?

Reply:
Hmm -- yes, just need to ensure that there are no issues with the `@abstract` `@property` not being implemented

Reply:
Nevermind, it's fine, I misunderstood ABC